src/wp-admin/includes/class-wp-site-health.php

```diff
@@ -2848,4 +2972,206 @@ public function is_development_environment() {
+			},
+			'age'                    => static function ( $header_value ) {
+				return is_numeric( $header_value ) && $header_value > 0;
+			},
+			'last-modified'          => '',
+			'etag'                   => '',
+			'x-cache'                => $cache_hit_callback,
+			'x-proxy-cache'          => $cache_hit_callback,
+			'cf-cache-status'        => $cache_hit_callback,
+			'x-kinsta-cache'         => $cache_hit_callback,
+			'x-aruba-cache'          => $cache_hit_callback,
+			'x-cache-enabled'        => static function ( $header_value ) {
+				return 'true' === strtolower( $header_value );
+			},
+			'x-cache-disabled'       => static function ( $header_value ) {
+				return ( 'on' !== strtolower( $header_value ) );
+			},
+			'cf-apo-via'             => static function ( $header_value ) {
+				return false !== strpos( strtolower( $header_value ), 'tcache' );
+			},
+			'x-srcache-store-status' => $cache_hit_callback,
+			'x-srcache-fetch-status' => $cache_hit_callback,
+			'cf-edge-cache'          => static function ( $header_value ) {
+				return false !== strpos( strtolower( $header_value ), 'cache' );
+			},
+		);
+	}
+
+	/**
+	 * Checks if site has page cache enabled or not.
+	 *
+	 * @since 6.1.0
+	 *
+	 * @return WP_Error|array {
+	 *     Page caching detection details or else error information.
+	 *
+	 *     @type bool    $advanced_cache_present        Whether a page caching plugin is present.
+	 *     @type array[] $page_caching_response_headers Sets of client caching headers for the responses.
+	 *     @type float[] $response_timing               Response timings.
+	 * }
+	 */
+	public function check_for_page_caching() {
+
+		/** This filter is documented in wp-includes/class-wp-http-streams.php */
+		$sslverify = apply_filters( 'https_local_ssl_verify', false );
+
+		$headers = array();
+
+		// Include basic auth in loopback requests. Note that this will only pass along basic auth when user is
+		// initiating the test. If a site requires basic auth, the test will fail when it runs in WP Cron as part of
+		// wp_site_health_scheduled_check. This logic is copied from WP_Site_Health::can_perform_loopback() in core.
+		if ( isset( $_SERVER['PHP_AUTH_USER'] ) && isset( $_SERVER['PHP_AUTH_PW'] ) ) { // phpcs:ignore WordPressVIPMinimum.Variables.ServerVariables.BasicAuthentication
+			// phpcs:ignore WordPress.Security.ValidatedSanitizedInput.InputNotSanitized, WordPressVIPMinimum.Variables.ServerVariables.BasicAuthentication
+			$headers['Authorization'] = 'Basic ' . base64_encode( wp_unslash( $_SERVER['PHP_AUTH_USER'] ) . ':' . wp_unslash( $_SERVER['PHP_AUTH_PW'] ) );
+		}
+
+		$caching_headers               = $this->get_page_cache_headers();
+		$page_caching_response_headers = array();
+		$response_timing               = array();
+
+		for ( $i = 1; $i <= 3; $i++ ) {
+			$start_time    = microtime( true );
+			$http_response = wp_remote_get( home_url( '/' ), compact( 'sslverify', 'headers' ) );
```

Comment:
Should this be `wp_safe_remote_get`?

Reply:
is it really needed? we are querying our own server.

Reply:
Per definition: This function is ideal when the HTTP request is being made to an arbitrary

Reply:
I think we should be fine with `wp_remote_get()`, since it's calling the site itself. Also almost all core usage is `wp_remote_get()` over `wp_safe_remote_get()`, so it's not a general policy either.